File: compiler/engine/testdata/reference/imported_fields/engine.ts
// TypeScript equivalent of imported_fields/engine.go.
import { Archetype, entityMemory, defineImport } from '@sonolus/sonolus.js-compiler/play'

const archetype = new Archetype('N')
const beat = defineImport('Beat')
const bpm = defineImport('Bpm')
const sum = entityMemory(Number)
defineImport(sum)

export function initialize() {
	sum.set(beat + bpm)
}

export function updateSequential() {
	sum.set(beat * 2 + bpm)
}

File: internal/compiler/engine/testdata/reference/imported_fields/engine.go
package p

type N struct {
	Beat float64 `sonolus:"imported"`
	Bpm  float64 `sonolus:"imported"`
	Sum  float64 `sonolus:"memory"`
}

func (n N) Initialize() {
	n.Sum = n.Beat + n.Bpm
}

func (n N) UpdateSequential() {
	n.Sum = n.Beat*2 + n.Bpm
}
